components/clp-package-utils/clp_package_utils/scripts/native/archive_manager.py

```diff
@@ -0,0 +1,365 @@
+    archives_dir = clp_config.archive_output.get_directory()
+    if not archives_dir.exists():
+        logger.error("`archive_output.directory` doesn't exist.")
+        return -1
+
+    if FIND_COMMAND == parsed_args.subcommand:
+        return _find_archives(
+            archives_dir,
+            database_config,
+            parsed_args.begin_ts,
+            parsed_args.end_ts,
+        )
+    elif DEL_COMMAND == parsed_args.subcommand:
+        if BY_IDS_COMMAND == parsed_args.del_subcommand:
+            return _delete_archives_by_ids(
+                archives_dir,
+                database_config,
+                parsed_args.ids,
+                parsed_args.dry_run,
+            )
+        elif BY_FILTER_COMMAND == parsed_args.del_subcommand:
+            return _delete_archives_by_filter(
+                archives_dir,
+                database_config,
+                parsed_args.begin_ts,
+                parsed_args.end_ts,
+                parsed_args.dry_run,
+            )
+
+
+def _find_archives(
+    archives_dir: Path,
+    database_config: Database,
+    begin_ts: int,
+    end_ts: int = None,
+) -> int:
+    """
+    Lists all archive IDs, if begin_its and end_its are provided,
+    only lists archives where `begin_ts <= archive.begin_timestamp` and
+    `archive.end_timestamp <= end_ts`.
+    :param archives_dir:
+    :param database_config:
+    :param begin_ts:
+    :param end_ts:
+    """
+    archive_ids: List[str]
+    logger.info("Starting to find archives from the database.")
+    try:
+        sql_adapter = SQL_Adapter(database_config)
+        clp_db_connection_params = database_config.get_clp_connection_params_and_type(True)
+        table_prefix = clp_db_connection_params["table_prefix"]
+        with closing(sql_adapter.create_connection(True)) as db_conn, closing(
+            db_conn.cursor(dictionary=True)
+        ) as db_cursor:
+
+            params = (begin_ts,)
+            query = f"SELECT id FROM `{table_prefix}archives` WHERE begin_timestamp >= %s"
+            if end_ts is not None:
+                query += " AND end_timestamp <= %s"
+                params = (begin_ts, end_ts)
+
+            db_cursor.execute(query, params)
+            results = db_cursor.fetchall()
+
+            archive_ids = [result["id"] for result in results]
+
+            if 0 == len(archive_ids):
+                logger.info("No archives found within specified time range.")
+                return 0
+
+            logger.info(f"Found {len(archive_ids)} archives within specified time range.")
+            for archive_id in archive_ids:
+                logger.info(archive_id)
+
+    except Exception:
+        logger.exception("Failed to find archives from the database.")
+        return -1
+
+    logger.info(f"Finished finding archives from the database.")
+
+    return 0
+
+
+def _delete_archives(
+    archives_dir: Path,
+    database_config: Database,
+    query: str,
+    params: List[str],
+    criteria: str,
```

Comment:
maybe we can name "criteria" as "command" instead, and refer to the defined constant?

We should also check whether the input command to this method is one of the expected value, and throw an error if it is not.

For params. My personal feeling is that it's tied to criteria, and passing it in as a separate argument could be confusing. What I can think of is to introduce a class that wraps around the param and criteria, but it might be an overkill.